subsys/net/lib/lwm2m/lwm2m_rd_client.c

```diff
@@ -117,16 +114,20 @@ static void set_sm_state(u8_t sm_state)
 	enum lwm2m_rd_client_event event = LWM2M_RD_CLIENT_EVENT_NONE;
 
 	/* Determine if a callback to the app is needed */
-	if (sm_state == ENGINE_BOOTSTRAP_DONE) {
-		event = LWM2M_RD_CLIENT_EVENT_BOOTSTRAP_COMPLETE;
-	} else if (client.engine_state == ENGINE_UPDATE_SENT &&
+#if defined(CONFIG_LWM2M_RD_CLIENT_SUPPORT_BOOTSTRAP)
+	if (sm_state == ENGINE_BOOTSTRAP_REG_DONE) {
+		event = LWM2M_RD_CLIENT_EVENT_BOOTSTRAP_REG_COMPLETE;
+	} else if (sm_state == ENGINE_BOOTSTRAP_TRANS_DONE) {
+		event = LWM2M_RD_CLIENT_EVENT_BOOTSTRAP_TRANSFER_COMPLETE;
+#endif
```

Comment:
Missing closing }

Reply:
Great find.  I wonder if I pulled in the wrong patches some how.. this never would have compiled when I was testing.